src/accelerate/utils/modeling.py

```diff
@@ -233,29 +237,36 @@ def values(self):
         return sum([x[1:] for x in self], [])
 
 
-def check_tied_parameters(model: nn.Module):
+def check_tied_parameters_in_config(model: nn.Module):
     """
     Check if there is any indication in the given model that some weights should be tied.
 
     Args:
-        model (nn.Module): The model to inspect
+        model (`torch.nn.Module`): The model to inspect
 
     Returns:
         bool: True if the model needs to have tied weights
     """
 
     # based on model.tie_weights() method
-    return any(
-        [
-            hasattr(model,"config")
+    has_tied_word_embedding = False
+    has_tied_encoder_decoder = False
+    has_tied_module = False
+
+    if transformers.modeling_utils.PreTrainedModel in inspect.getmro(model.__class__):
```

Comment:
I was thinking on testing the class `__name__` to avoid the extra dep on Transformers.